### DISCUSSION TITLE
Customizing environment variables before preview deploys?

### DISCUSSION THREAD

**Discussion by rohan** [vercel/community#688] [Help] [answered]
Hi folks! I have a Next.js project hosted on Vercel that talks to an API hosted on a different provider. I have that other provider configured to spin up preview builds with their own domains, and I have an environment variable in Next that sets the backend URL. When Vercel deploys my PRs, I'd like to set the environment variable to the appropriate preview build URL (which, if it helps, has a fixed format). Is there an easy way to set the environment variable *before* the Vercel build kicks off, so that I don't have to wait for the first deploy, set the variable, and then redeploy?

**Comment by amyegan** [comment, marked as answer]
Hi @rohan. If I'm understanding your workflow correctly, I think you could use the [Ignored Build Step](https://vercel.com/docs/concepts/projects/overview#ignored-build-step) setting to skip the automatic preview deployment and then deploy using [Vercel CLI](https://vercel.com/docs/cli) after setting your [environment variable](https://vercel.com/docs/cli#commands/env).

**Reply by rohan** [reply]
Hey @amyegan — thanks, that makes sense! Quick follow-up: when I set my Ignored Build step to, e.g., `git diff --quiet main HEAD -- file` (i.e., checking that `file` was unchanged), I get an error saying that `main` doesn't exist. I've set it to compare `HEAD^` to `HEAD`, but I'd like to check that no commit in the branch changed `file`. How can I do that?

**Reply by amyegan** [reply]
If you want to only deploy when a certain file has changed, you could do something like `git diff HEAD^ HEAD --quiet ./file`. If you want to only deploy production builds, you could set the command to `[ "$VERCEL_ENV" != production ]` which ignores commits all branches aside from your production branch and skips PR previews.

If you want finer control over the conditions for automatic deployment of specific branches then a script is your best option. You can see out more details on the different Ignored Build Step options here https://vercel.com/support/articles/how-do-i-use-the-ignored-build-step-field-on-vercel
